pkg/gas/models.go

```diff
@@ -362,13 +362,23 @@ func (e *evmFeeEstimator) estimateFeeLimit(ctx context.Context, feeLimit uint64,
 	if !e.geCfg.EstimateLimit() {
 		return providedGasLimit, nil
 	}
+
+	// If EstimationSenderAddress string is not empty, then we unwrap the config value and set the pointer. Otherwise, it remains nil
+	var estimationSenderAddress *common.Address
+	if e.geCfg.EstimationSenderAddress() != nil && len(*e.geCfg.EstimationSenderAddress()) != 0 {
+		estimationSenderAddressUnwrapped := common.HexToAddress(*e.geCfg.EstimationSenderAddress())
+		estimationSenderAddress = &estimationSenderAddressUnwrapped
+	}
+
 	// Create call msg for gas limit estimation
 	// Skip setting Gas to avoid capping the results of the estimation
 	callMsg := ethereum.CallMsg{
 		To:   toAddress,
 		Data: calldata,
 	}
-	if fromAddress != nil {
+	if estimationSenderAddress != nil {
+		callMsg.From = *estimationSenderAddress
```

Comment:
Sorry to add another nit but looking at this a little longer I realized we're doing the same check twice. Could we just replace this with something like below and drop `estimationSenderAddress` 
```suggestion
	if e.geCfg.EstimationSenderAddress() != nil && len(*e.geCfg.EstimationSenderAddress()) != 0 {
		callMsg.From = common.HexToAddress(*e.geCfg.EstimationSenderAddress())
```

Reply:
Thank you Amit; this makes sense, and I've just updated the PR to reflect this